dll/shellext/zipfldr/CZipExtract.cpp

```diff
@@ -95,6 +106,42 @@ class CZipExtract :
 
         int OnWizardNext()
         {
+            if (m_hExtractionMutex != INVALID_HANDLE_VALUE)
+            {
+                WaitForSingleObject(m_hExtractionMutex, INFINITE);
+                if (m_bExtractionThreadOver)
```

Comment:
Since this variable is initialized to `true`, you'd enter this path even if thread creation fails, then the user clicks `Next` again. So you probably want to initialize it to `false`.

Reply:
I'll make a pass later to see if I can simplify the whole boolean fest.

Reply:
Also, having SetActivePage implemented could avoid having to re-enter OnWizardNext after the extraction (and probably get rid of m_bExtractionThreadSuccess): https://docs.microsoft.com/en-us/cpp/mfc/reference/cpropertysheet-class?view=vs-2019#setactivepage